flask-server.py

```diff
@@ -25,122 +26,167 @@
-            cal_event.add('description', event['description'])
-            
-            location = event['location']
-            location_str = ', '.join(filter(None, [
-                location.get('building'),
-                location.get('room_number'),
-                location.get('street'),
-                location.get('city'),
-                location.get('state'),
-                location.get('zipcode')
-            ]))
-            cal_event.add('location', location_str)
-            cal.add_component(cal_event)
+            cal_event.name = event["title"]
+            cal_event.begin = event["start"]
+            cal_event.end = event["end"]
+            cal_event.description = event["description"]
+
+            location = event["location"]
+            location_str = ", ".join(
+                filter(
+                    None,
+                    [
+                        location.get("building"),
+                        location.get("room_number"),
+                        location.get("street"),
+                        location.get("city"),
+                        location.get("state"),
+                        location.get("zipcode"),
+                    ],
+                )
+            )
+            if location_str:
+                cal_event.location = location_str
+
+            cal.events.add(cal_event)
         except Exception as e:
             print(e)
-        
-    # Create a response object
-    response = Response(cal.to_ical(), mimetype='text/calendar')
-    response.headers['Content-Disposition'] = 'attachment; filename=calendar.ics'
-    
+
+    response = Response(str(cal), mimetype="text/calendar")
+    response.headers["Content-Disposition"] = "attachment; filename=calendar.ics"
     return response
 
+
+@app.route("/v0/public.ics/<event_id>")
+def get_event(event_id):
+    try:
+        # Find first event with matching id
+        event = next(e for e in alldata if e["id"] == event_id)
+        cal = Calendar()
+        cal_event = Event()
+        cal_event.name = event["title"]
+        cal_event.begin = event["start"]
+        cal_event.end = event["end"]
+        cal_event.description = event["description"]
+
+        location = event["location"]
+        location_str = ", ".join(
+            filter(
+                None,
+                [
+                    location.get("building"),
+                    location.get("room_number"),
+                    location.get("street"),
+                    location.get("city"),
+                    location.get("state"),
+                    location.get("zipcode"),
+                ],
+            )
+        )
+        if location_str:
+            cal_event.location = location_str
+
+        cal.events.add(cal_event)
+
+        response = Response(str(cal), mimetype="text/calendar")
+        response.headers["Content-Disposition"] = f"attachment; filename={event_id}.ics"
+        return response
+    except StopIteration:
```

Comment:
I'm not sure ive ever seen this exception type before. Do you have documentation on what it is used for or where it comes from?